src/atoms/TextComponents/TextComponent.js

```diff
@@ -0,0 +1,56 @@
+import React, { PropTypes } from 'react';
+import styles from './textComponent.module.scss';
+import classnames from 'classnames';
+
+function TextComponent ( props ) {
+  const {
+    tag,
+    children,
+    type,
+    className,
+    semibold,
+    light
+  } = props;
+
+  return React.createElement(
+    tag,
+    { className: classnames(
+      className,
+      styles['typography-' + type],
+      semibold ? styles.semibold : styles.regular,
```

Comment:
this would require you to explicitly pass `light={false}`, otherwise the binary below will add `styles.regular` and override the semibold styling. maybe there could be a `weight` prop which is optional and defaults to regular, with semibold & regular as the type choices